libraries/AP_ExternalAHRS/AP_ExternalAHRS_InertialLabs.h

```diff
@@ -0,0 +1,226 @@
+    };
+
+    struct PACKED vec3_16_t {
+        int16_t x,y,z;
+        Vector3f tofloat(void) {
+            return Vector3f(x,y,z);
+        }
+    };
+    struct PACKED vec3_32_t {
+        int32_t x,y,z;
+        Vector3f tofloat(void) {
+            return Vector3f(x,y,z);
+        }
+    };
+    struct PACKED vec3_u8_t {
+        uint8_t x,y,z;
+        Vector3f tofloat(void) {
+            return Vector3f(x,y,z);
+        }
+    };
+    struct PACKED vec3_u16_t {
+        uint16_t x,y,z;
+        Vector3f tofloat(void) {
+            return Vector3f(x,y,z);
+        }
+    };
+
+    struct gnss_extended_info_t {
+        uint8_t fix_type;
+        uint8_t spoofing_status;
+    };
+
+    struct gnss_info_short_t {
+        uint8_t info1;
+        uint8_t info2;
+    };
+    
+    union PACKED ILabsData {
+        uint32_t gps_time_ms; // ms since start of GPS week
+        uint16_t gps_week;
+        vec3_32_t accel_data_hr; // g * 1e6
+        vec3_32_t gyro_data_hr; // deg/s * 1e5
+        struct PACKED {
+            uint16_t pressure_pa2; // Pascals/2
+            int32_t baro_alt; // meters*100
+        } baro_data;
+        vec3_16_t mag_data; // nT/10
+        struct PACKED {
+            int16_t yaw; // deg*100
+            int16_t pitch; // deg*100
+            int16_t roll; // deg*100
+        } orientation_angles; // 321 euler order?
+        vec3_32_t velocity; // m/s * 100
+        struct PACKED {
+            int32_t lat; // deg*1e7
+            int32_t lon; // deg*1e7
+            int32_t alt; // m*100, AMSL
+        } position;
+        vec3_u8_t kf_vel_covariance; // mm/s
+        vec3_u16_t kf_pos_covariance;
+        uint16_t unit_status;
+        gnss_extended_info_t gnss_extended_info;
+        uint8_t num_sats;
+        struct PACKED {
+            int32_t hor_speed; // m/s*100
+            uint16_t track_over_ground; // deg*100
+            int32_t ver_speed; // m/s*100
+        } gnss_vel_track;
+        uint32_t gnss_pos_timestamp; // ms
+        gnss_info_short_t gnss_info_short;
+        uint8_t gnss_new_data;
+        uint8_t gnss_jam_status;
+        int32_t differential_pressure; // mbar*1e4
+        int16_t true_airspeed; // m/s*100
+        vec3_16_t wind_speed; // m/s*100
+        uint16_t air_data_status;
+        uint16_t supply_voltage; // V*100
+        int16_t temperature; // degC*10
+        uint16_t unit_status2;
+    };
+
+    AP_ExternalAHRS::gps_data_message_t gps_data;
+    AP_ExternalAHRS::mag_data_message_t mag_data;
+    AP_ExternalAHRS::baro_data_message_t baro_data;
+    AP_ExternalAHRS::ins_data_message_t ins_data;
+    AP_ExternalAHRS::airspeed_data_message_t airspeed_data;
+
+    uint16_t buffer_ofs;
+    uint8_t buffer[256]; // max for normal message set is 167+8
+
+private:
+    AP_HAL::UARTDriver *uart;
+    int8_t port_num;
+    uint32_t baudrate;
+    bool setup_complete;
+
+    void update_thread();
+    bool check_uart();
+    bool check_header(const ILabsHeader *h) const;
```

Comment:
```suggestion
    bool check_header(const ILabsHeader *h) const WARN_IF_UNUSED;
```

Reply:
i'm not a fan of cluttering internal APIs like that

Reply:
Ah, yea, it's internal. Disregard.